### DISCUSSION TITLE
SINDYc - SR3 - Thresholds getting disregarded

### DISCUSSION THREAD

**Discussion by IamKFJ** [dynamicslab/pysindy#399] [General]
I am trying to set coefficient thresholds by using either the "threshold" variable for an overall threshold or using the "thresholds" variable to set each threshold individually.
But when running my code I frequently get values lower than my threshold.

Is this a known Problem?
Or am I misunderstanding how thresholds work?

This is my code:

`
def SINDYc(X_Data=None, U_Data=None, dT=None, opt_threshold=0.01, **kwargs):

    dT = dT
    feature_names = ['x', 'v', 'F', 'u']

    model = SINDy(feature_names = feature_names, optimizer=ps.SR3(threshold=opt_threshold)) 
    model.fit(x=X_Data, t=dT, u=U_Data)

    model.print()

   return model.equations()
`

I set 'opt_threshold' to 0.013 and regularly get coefficient values smaller than that (e.g. 0.006 or -0.001) 

Example output:
`(x)' = 1.000 v
(v)' = -9.810 1 + -0.015 x + 0.333 F + -0.034 v^2
(F)' = 0.006 v + -0.001 F + 1.001 u`

Am I missing something?
I would appreciate any help or guidance on how to properly use thresholds.

**Comment by znicolaou** [comment]
Hello, thresholds are applied to normalized features by default (which is admittedly a little confusing but usually leads to better fits). Try fitting with `optimizer=ps.SR3(threshold=opt_threshold,normalize_columns=False)` to see if you get your expected results.
